src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -293,6 +293,54 @@ EnsureReferenceTablesExistOnAllNodesExtended(char transferMode)
 }
 
 
+bool
+HasNodesWithMissingReferenceTables(List **referenceTableList)
+{
+	int colocationId = GetReferenceTableColocationId();
+
+	if (colocationId == INVALID_COLOCATION_ID)
+	{
+		/* we have no reference table yet. */
+		return false;
+	}
+	LockColocationId(colocationId, AccessShareLock);
+
```

Comment:
minor: seems like this function could be shorter:
```c
	int colocationId = GetReferenceTableColocationId();
	if (colocationId == INVALID_COLOCATION_ID)
	{
		/* we have no reference table yet. */
		return false;
	}
	
	LockColocationId(colocationId, AccessShareLock);

    List *workerNodeList = ReferenceTablePlacementNodeList(AccessShareLock);

    WorkerNode *workerNode = NULL;
    foreach_ptr(workerNode, workerNodeList)
    {
        if (!NodeHasAllReferenceTableReplicas(workerNode))
        {
            return true;
        }
    }

    return false;
```

Reply:
This would load the reference table list repeatedly for all nodes. Both nodes and the number of reference table could be quite sizeable. I think.